### PR TITLE
Jbpm tests stabilisation

### PR DESCRIPTION
**Pull request by dcihak**
AsyncAuditLogProducerTest test fixed: nodes and process instances logging

### CODE REVIEW

**Comment by dcihak** [comment]
READY TO BE MERGED

**Comment by tsurdilo** [comment]
My only comment here is that I'm not big fan of mixing assertj and junit in a test. Either do one or the other all the way. Maybe we could remove junit and replace its assertions with assertj.

**Comment by rsynek** [comment]
@tsurdilo @dcihak +1, let's move to assertj whenever we touch some test (doing a global big bang is probably too much).

File: jbpm-audit/src/test/java/org/jbpm/process/audit/jms/AsyncAuditLogProducerTest.java [hunk 1]
```diff
@@ -484,7 +482,7 @@ public void onMessage(Message message) {
             };
             consumer.setMessageListener(rec);
             // since we use message listener allow it to complete the async processing
-            Thread.sleep(2000);
+            Thread.sleep(waitTime);
```
Comment (rsynek) [inline jbpm-audit/src/test/java/org/jbpm/process/audit/jms/AsyncAuditLogProducerTest.java]: Although I like parameterization of Thread.sleep, I can see the waiting raised from 2 secs to 5. Is this the only way to make the tests stable? It prolongs their execution.

**Comment by MarianMacik** [comment]
@mswiderski Is there any chance to check if there are still any JMS messages left to be consumed? So we can e.g. constantly check the number and don't have to use Thread.sleep(). If there is no good and quick solution for it (HornetQMessageConsumer does not have it), we can leave it as is.

By looking at the JMS documentation, the only provider-independent way to check a number of messages in a Queue is to use QueuBrowser and simply every time count number of messages in it, which is not convenient.

**Comment by dcihak** [comment]
Using of junit methods replaced by assertj

**Comment by dcihak** [comment]
Thread.sleep() was replaced by latch countdown to wait until all the messages in the queue are processed.

**Pull request merged**